cou/cli.py

```diff
@@ -42,110 +71,333 @@ def parse_args(args: Any) -> argparse.Namespace:
+    )
+
+    # Arg parser for "cou help" sub-command
+    help_parser = subparsers.add_parser(
+        "help",
+        usage="cou help [command]",
+    )
+    help_parser.add_argument(
+        "subcommand",
+        nargs="?",
+        choices=["plan", "run", "all"],
+        default="all",
+        type=str,
+        help="A sub-command to get information of.",
+    )
+
+    # Define common options for subcommands and their child commands
+    subcommand_common_opts_parser = argparse.ArgumentParser(add_help=False)
+    subcommand_common_opts_parser.add_argument(
         "--model",
         default=None,
         dest="model_name",
+        type=str,
         help="Set the model to operate on. If not set it gets the model name in this order:\n"
         "  1 - Environment variable JUJU_MODEL,"
         "  2 - Environment variable MODEL_NAME,"
         "  3 - Current active juju model",
     )
-    parser.add_argument(
+    subcommand_common_opts_parser.add_argument(
+        "--parallel",
+        help="Run upgrade steps in parallel where possible.",
+        default=False,
+        action="store_true",
+    )
+
+    # quiet and verbose options are mutually exclusive
+    group = subcommand_common_opts_parser.add_mutually_exclusive_group()
+    group.add_argument(
+        "--verbose",
+        "-v",
+        default=0,
+        action="count",
+        dest="verbosity",
+        help="Increase logging verbosity in STDOUT. Repeat the 'v' in the short option "
+        "for more detail. Maximum verbosity is obtained with 4 (or more) "
+        "v's, i.e. -vvvv. \nNote that this doesn't affect the verbosity in logfile, "
+        "which will always have the maximum verbosity.",
+    )
+    group.add_argument(
+        "--quiet",
+        "-q",
+        default=False,
+        action="store_true",
+        dest="quiet",
+        help="Disable output in STDOUT.",
+    )
+
+    # Define options specific to data-plane child command and make them mutually exclusive
+    dp_subparser = argparse.ArgumentParser(add_help=False)
+    dp_upgrade_group = dp_subparser.add_mutually_exclusive_group()
+    dp_upgrade_group.add_argument(
+        "--machine",
+        "-m",
+        action="append",
+        help="Specify machines ids to upgrade.",
+        dest="machines",
+    )
+    dp_upgrade_group.add_argument(
+        "--hostname",
+        "-n",
+        action="append",
+        help="Specify machine hostnames to upgrade.",
+        dest="hostnames",
+    )
+    dp_upgrade_group.add_argument(
+        "--availability-zone",
+        "--az",
+        action="append",
+        help="Specify availability zones to upgrade.",
+        dest="availability_zones",
+    )
+
+    # Arg parser for "cou plan" sub-command
+    plan_parser = subparsers.add_parser(
+        "plan",
+        description="Show the steps for upgrading the cloud to the next release.",
+        help="Show the steps for upgrading the cloud to the next release.",
```

Comment:
I want to avoid giving the impression we'd print juju cli commands
```suggestion
        description="Show the steps COU will take to upgrade the cloud to the next release.",
        help="Show the steps COU will take to upgrade the cloud to the next release.",
```